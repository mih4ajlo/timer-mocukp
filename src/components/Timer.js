import react from 'react';

import React, { Component } from 'react';

class Timer extends Component {
    

    constructor(props) {
        super(props);
        this.state = { 
            seconds: props.basicTime /*7158*/,
            pause:false,
            byoyomiPeriods:3,
            byoyomi:false,
            };


        this.pause = this.pause.bind(this);    
    }

    tick(){

        this.setState(state=>{
            
            if( !state.pause )
            
            return {seconds : state.seconds - 1,}
            
        }) ;
    }

    pause(){

        this.setState( state => ({ pause : !state.pause }) )

    }

    componentDidMount(){
        this.interval = setInterval( () => this.tick(), 1000);
    }

    componentWillUnmount() {
        clearInterval( this.interval );
    }

    render() {

                    //~~ == Math.floor
        let hours = Math.floor(this.state.seconds / 3600);



        let minutes =  Math.floor( (this.state.seconds - hours * 3600) / 60);
        let seconds =  this.state.seconds - minutes * 60 - hours * 3600;

        //prvo da idemo u nazad
        
        if(this.state.seconds == 0){
            //treba da se stopira, ako je istekao byoyomi, 
            //ili ako je sd (nema byoyomija)
            if(this.props.byoyomi){
                alert(this.props.byoyomiTime)
            }
            clearInterval( this.interval );
        }


        return (
            <div>
                <span> hours:{hours}, minutes:{minutes}, seconds: {seconds} </span>
                <div>ovde da ide cela povrsina; podeljeno na dva ekrana, da se klikne</div>
                <div> <button onClick={this.pause}>Pause</button> </div>
            </div>
        );
    }
}

export default Timer;
