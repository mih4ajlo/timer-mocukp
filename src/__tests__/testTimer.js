import React from 'react';
import {create} from 'react-test-renderer';
import Timer from '../components/Timer';

	const timerRef = React.createRef();

	const component = create(<Timer 

		ref={timerRef}
		key={0}

		playerState =  {{

				seconds:  300 ,
				byoyomiPeriods: 3,
				byoyomiTime: 10,
				moveNo:0,
				pause:false,

	    	}}

		tick1 = { ()=>{console.log("thick fn")} }

		countAll={ ()=>console.log("this.countAll")}
		move={ ()=> console.warn('move fn log') }
        		

	/>);

	const instance  = component.getInstance();

describe("Timer component",()=>{
	test("it matches snapshot",()=>{


		expect(component.toJSON()).toMatchSnapshot();
	})
})

//ovo treba da fail-uje ako se ne proslede parametri

describe("Timer component",()=>{
        test("text testing",()=>{
        
		

		expect(instance.state.seconds).toBe(300);
        })
})




describe("Timer component",()=>{
        test("click testing",()=>{
        
		instance.pauseProxy();

		expect(instance.state.seconds).toBe(300);
        })
})

