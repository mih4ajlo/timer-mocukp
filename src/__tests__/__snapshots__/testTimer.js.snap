// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer component it matches snapshot 1`] = `
<div
  style={null}
>
  <div>
    Broj poteza 
    0
  </div>
  <span>
     hours:
    0
    , minutes:
    5
    , seconds: 
    0
     
  </span>
  <div>
    byoyomi periods: 
    3
    , byoyomi time: 
    10
  </div>
  <div>
    ovde da ide cela povrsina; podeljeno na dva ekrana, da se klikne
  </div>
  <div>
     
    <button
      onClick={[Function]}
    >
      Move
    </button>
     
  </div>
</div>
`;
